tests/test_columns.py

```diff
@@ -635,18 +659,16 @@ def test_cast_to_data_type(dtype):
     )
 
 
-@pytest.mark.parametrize(
-    "dtype", ["string", "timestamp", "date", "datetime", "decimal(18, 0)"]
-)
-def test_nested_cast_to_data_type(dtype):
-    sql = f"""INSERT OVERWRITE TABLE tab1
+@pytest.mark.parametrize("dtype", ["string", "timestamp", "date", "decimal(18, 0)"])
+def test_nested_cast_to_data_type(dtype: str):
```

Comment:
Can you take a look at this test case? It's extremely slow to run.

Reply:
```python
def test_nested_cast_to_data_type():
    dtype = "string"
    sql = f"""INSERT INTO tab1
SELECT cast(cast(cast(cast(cast(col1 AS {dtype}) AS {dtype}) AS {dtype}) AS {dtype}) AS {dtype}) AS col1
FROM tab2"""
    assert_column_lineage_equal(
        sql,
        [(ColumnQualifierTuple("col1", "tab2"), ColumnQualifierTuple("col1", "tab1"))],
        test_sqlparse=False
    )
```

I tried this, and it's 1 sec 534 ms to run on my machine. Doesn't look like a dialect issue.

Reply:
OK, this is sqlfluff's issue. Won't be a blocker for this PR then. Let's resolve these. Maybe I'll try if I can submit a PR to them later.